README.md

```diff
@@ -41,231 +41,383 @@ This project contains two separate providers that operate in tandem:
 - kubectl version v1.11.3+.
 - Access to a Kubernetes v1.11.3+ cluster.
 
-### Installation
-To configure clusterctl with the OpenShift Agent providers, edit `~/.cluster-api/clusterctl.yaml` and add the following:
+# Usage
+
+## Prerequisites
+
+1. **Cluster Environment**:
+  - A bootstrap Kubernetes cluster (e.g., a `kind` instance).
+  - Ensure `kubectl` is configured to interact with the cluster.
+  - optional:
+
+2. **Pull Secret (OCP only)**:
+  - For OCP, ensure you have a valid pull secret. This is not required for OKD.
+  - The pull secret can be retrieved from [Red Hat OpenShift Console](https://console.redhat.com/openshift/install/pull-secret).
+
```

Comment:
Need to mention the dns resolve requirements: 3 entries for the api, api-int, .apps; 2 for the assisted-service and assisted-image